snapcraft/internal/elf.py

```diff
@@ -36,6 +36,49 @@
 logger = logging.getLogger(__name__)
 
 
+class Library:
+    """Represents the SONAME and path to the library."""
+
+    def __init__(self, *, soname: str, path: str, root_path: str,
+                 core_base_path: str) -> None:
+        self.soname = soname
+        if path.startswith(root_path) or path.startswith(core_base_path):
+            self.path = path
+        else:
+            self.path = _crawl_for_path(soname=soname,
+                                        root_path=root_path,
+                                        core_base_path=core_base_path)
+        # Required for libraries on the host and the fetching mechanism
+        if not self.path:
+            self.path = path
+
+        system_libs = _get_system_libs()
+        if soname in system_libs:
+            self.system_lib = True
+        else:
+            self.system_lib = False
+
+        if path.startswith(core_base_path):
+            self.in_base_snap = True
+        else:
+            self.in_base_snap = False
+
+
+@lru_cache()
```

Comment:
This is new to me, pretty sweet! This doesn't really seem to fit into the "most recent calls are the best predictors of upcoming calls" mold, but I doubt this hurts. Would be interesting to see a benchmark.

Reply:
they probably aren't but given the size of the default cache (128) I thought it was good enough (better than constant calls).